### PR TITLE
Update Newspack Blocks

### PR DESCRIPTION
**Pull request by adekbadek**
#### Changes proposed in this Pull Request

Updates Newspack Blocks to [v1.30.0](https://github.com/Automattic/newspack-blocks/releases/tag/v1.30.0).

https://github.com/Automattic/newspack-blocks/pull/725 changes how deduplication is handled.

Post Carousel also has several new options that affect how it renders on the front-end.

#### Testing instructions

1. Ensure the new version of `newspack-blocks` is synchronised with FSE
2. Create a page and insert both Blog Posts and Carousel blocks
3. Observe that the deduplication works across all blocks - no posts are duplicated
4. For the Post Carousel, observe the new options and confirm that they work as expected in the editor and on the front-end:

Screen Shot 2021-07-19 at 10 39 28 PM

  - Slide aspect ratio
  - Image fit
  - Number of slides to show at once
  - Post types (should allow posts and pages, at least, unless the site has plugins that register additional post types for the block)

### CODE REVIEW

**Comment by dkoo** [comment]
@adekbadek Picking up the thread from https://github.com/Automattic/newspack-blocks/pull/725.

I merged the latest `trunk` into this branch and synced the latest version of `newspack-blocks` code to my sandbox with updated paths. Now, the Blog Posts block works fine, but the Posts Carousel block crashes hard as soon as it's inserted:

Screen Shot 2021-04-06 at 11 18 30 AM

As you can see in the screenshot, the placeholder files in `/assets` are not being referenced from the right location either, but I haven't dug in yet to where they should go.

**Comment by adekbadek** [comment]
@dkoo I've rebased this branch onto `trunk`. When testing _this_ branch of Calypso and syncing with the Blocks' `master`, I see no issues: a Homepage Posts block & Carousel block render fine in both editor and on the front-end, and have shared de-duplication. Could you please check again?

**Comment by dkoo** [comment]
@adekbadek I can confirm that I too no longer see the error described in https://github.com/Automattic/wp-calypso/pull/51583#issuecomment-814293097, and that shared deduplication is working across Blog Posts and Post Carousel blocks. 🥳 

However, the block attributes for Post Carousel seem to be breaking the SwiperJS functionality in the editor preview for me. This is when syncing the latest `master` of newspack-blocks to my sandbox. I'll see if I can find out more info on why.

EDIT: This is a bug in the Blocks repo. Fix is here: https://github.com/Automattic/newspack-blocks/pull/807

**Comment by adekbadek** [comment]
With the fix on Blocks ready, all that's left in this PR is the hardcoded version bump after the fix is released, right?

**Comment by dkoo** [comment]
Tested in both sandbox and Atomic environments. edb0513 updates outdated file paths in the sync script, which caused the plugin to be missing files and throw fatals in the latter environment.

**Comment by dkoo** [comment]
> To test your changes on WordPress.com, run `install-plugin.sh notifications add/newspack-blocks-update` on your sandbox.

I ran this on my sandbox and tested the updated build successfully.

**Comment by dkoo** [comment]
Rebased on `trunk` and re-tested in both sandbox and Atomic. Everything looks 👍 to me, so I think this is ready for a merge, now!

File: apps/editing-toolkit/editing-toolkit-plugin/newspack-blocks/carousel-block-editor.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
 import { registerBlockType } from '@wordpress/blocks';
 import { addFilter } from '@wordpress/hooks';
+import { CAROUSEL_BLOCK_NAME } from './consts';
 import { settings } from './synced-newspack-blocks/blocks/carousel/index';
 
 /**
- * Block name in the A8C\FSE context.
+ * Internal dependencies
  */
```
Comment (noahtallen) [inline apps/editing-toolkit/editing-toolkit-plugin/newspack-blocks/carousel-block-editor.js]: I think this entire comment could be deleted. At least, "internal dependencies" seems out of place :)

**Review by noahtallen — APPROVED** [review]
The changes here look good to me. The only thing a bit odd is the addition of @wordpress/elements v3, though since all the builds pass (and smoke tests work well), this should not be problematic. It originates in the new dependency on @wordpress/icons in newspack blocks.

I'll note that I didn't test the newspack changes on my sandbox though. @dkoo I think a new version of Gutenberg was released to wpcom in the past 12 hours. I think it's live for simple sites, but not yet for Atomic. It would probably be worth re-testing it in those environments :)

File: apps/editing-toolkit/editing-toolkit-plugin/newspack-blocks/carousel-block-editor.js (hunk 1 above)
Comment (dkoo) [reply apps/editing-toolkit/editing-toolkit-plugin/newspack-blocks/carousel-block-editor.js]: Removed in a1caf1c. Thanks for the catch!

**Pull request merged**